### ISSUE TITLE
Request: "Blue" Social Status

### ISSUE THREAD

**Issue by GearBell** [Frooxius/NeosPublic#715]
Lots of people stay in private worlds for various reasons, often we cant tell if they do not want to be disturbed or if they simply want to "hide" - can we have a Blue social status much like vrchat so those who dont mind visitors can signal it in the social tab? (We currently have red, orange, green, grey) Blue means "Its okay to join me! Come on over!" At the moment it could just be a new color ring in the social tab, merely an indication to friends who is welcoming to join or not. [Though an added functionality of blue status is that friend is automatically joinable regardless of an invite. Setting yourself to blue is consenting to letting friends in.] For now a simple color indicator would be nice. (Everyones hiding in private worlds! Cant tell who is okay to join or not.)

**Comment by shiftyscales** [comment]
There are plenty of reasons folks use private worlds- there are some planned features in place that could help promote keeping more open sessions, e.g. adding tags to users to have allow/block lists that add more granularity to the existing privacy settings. You've also previously requested Friends+ as per #701 which could help here too.

I feel rather than having another designated status, it could perhaps be better that the fifth status is "custom" sort of like in Discord. So you could have your own custom status string. "Working on Logix.", "Watching a movie.", "Join me!", etc.

The existing interface for it, like every other part of the interface will be reworked.

The custom status would also allow a much wider range of uses (and could even highly likely deprecate the existing 'Away' status since you could type an away message that was more specific. "Away for 20.", "Getting lunch." etc.

**Comment by GearBell** [comment]
custom status, definatly sounds like a great idea! (itll still need a distinct color though) but yes i like how versatile that can be! However, i garantee people are gonna use it like a facebook status though....a useful "working on logix, visitors welcome" versus "pickles are lame" status cause theyll think its funny. vrchat did the custom statuses, 3/10 times it was an actual status, the rest were either lewd, rude, or stupid. Being that its neos though, it would be 6/10 useful (because we are more mature than vrchat) - the other 4/10s will probably use them for relationship statuses or something... [solution? perhaps preset custom statuses?]

**Comment by Frooxius** [comment]
Yeah custom status is definitely coming. That's a good point about having it reset after a while so it's not used for permanent stuff. When exactly would it be reset though? At restart? Perhaps after Neos has been off for a while, so if the user is just briefly restarting Neos, it stays.

For the blue one, I'm not entirely sure, as it would make the green "Online" into essentially "Busy, do not disturb" one.

I wouldn't definitely tie the joinability to the status itself, that's better just configured through the world access settings, those will be expanded as well.

**Comment by GearBell** [comment]
A reset sounds like a good idea - how about after 30 minutes offline? A status would be nice to see in the friends list so we can quickly see as we scroll, if not a different color, then how about a double ring? that way they can be any status color, but a double ring means they have a custom status.

**Comment by Frooxius** [comment]
I think we could do a custom color for the custom one. Maybe even have a few colors to pick from? But that might get too confusing if it's too many, as there won't be any commonly known association with what the colors mean.

**Comment by GearBell** [comment]
i think too many custom colors would be very confusing and just as easily over customized as the "pickles are lame" excuse. It really should only be one color. The only colors on the spectrum on the wheel left unused by the current statuses are blue. purple is used for freind request, pink for Different Version, green/yellow(orange?)/grey/red is already current statuses. I wasnt suggesting blue before because "its a pretty color", its cause its the only unused one at the moment it looks like. If colors are too confusing, the double ring thing might make more sense? that way it can layer on top of the current statuses. "Red" but "OK to contact", Orange but "At Keyboard, Poke if Needed", Green but "Super tired, dont ask 4 logix help rn", etc.

**Comment by TehTurk** [comment]
I agree for using blue  for custom, custom colors sound fun but it's also confusing after awhile too. Emission color could also help too, or even maybe use something of a mask/stripe/gradient to indicate it's a status but custom type of status. It doesn't have to be limited to just color.

**Comment by ukilop** [comment]
> It doesn't have to be limited to just color.
Why not shape then? Square status maybe?

**Comment by shiftyscales** [comment]
It's not really worth considering the appearance at all, @ukilop. By the nature of the facet system, All of the UI can have its visuals customized to suit the desires of the user. What matters more is considering the underlying functionality, and how the feature would be designed, and implemented on a technical level.
